frontend/src/composables/useCustomColors.js

```diff
@@ -15,10 +18,14 @@ import { get } from '@/lodash'
 
 export const useCustomColors = (customThemes, theme = useTheme()) => {
   return new Promise((resolve, reject) => {
+    const timeoutId = setTimeout(() => {
+      unwatch()
+      reject(new Error('Setting custom colors timed out'))
+    }, 3000)
     const unwatch = watch(customThemes, value => {
       if (value) {
         clearTimeout(timeoutId)
-        unwatch()
+        nextTick(() => unwatch())
```

Comment:
unrelated to this PR you might want to return/continue early to avoid deep if nestings

```js
export const useCustomColors = (customThemes, theme = useTheme()) => {
  return new Promise((resolve, reject) => {
    const timeoutId = setTimeout(() => {
      unwatch()
      reject(new Error('Setting custom colors timed out'))
    }, 3000)
    const unwatch = watch(customThemes, value => {
      if (!value) { // <--------- here
        return
      }

      clearTimeout(timeoutId)
      nextTick(() => unwatch())
      for (const colorMode of ['light', 'dark']) {
        const vuetifyThemeColors = toValue(theme.themes)?.[colorMode]?.colors ?? {}
        const customThemeColors = value[colorMode] ?? {}
        for (const [key, value] of Object.entries(customThemeColors)) {
          if (!(key in vuetifyThemeColors)) { // <--------- here
            continue
          }

          const colorCode = get(vuetifyColors, value)
          if (colorCode) {
            vuetifyThemeColors[key] = colorCode
          } else if (isHtmlColorCode(value)) {
            vuetifyThemeColors[key] = value
          }
        }
      }
      resolve()
    }, {
      immediate: true,
    })
  })
}
```

Reply:
✔️